htap.go

```diff
@@ -19,8 +27,31 @@ func (qcc *queryContextCache) init() *queryContextCache {
 	return qcc
 }
 
-func (qcc *queryContextCache) add(qces ...queryContextEntry) {
+func (qcc *queryContextCache) add(sc *snowflakeConn, qces ...queryContextEntry) {
 	qcc.mutex.Lock()
 	defer qcc.mutex.Unlock()
-	qcc.entries = append(qcc.entries, qces...)
+	if len(qces) == 0 {
+		qcc.prune(0)
```

Comment:
it's not really adding when we remove all entries from the list, if I understand it correctly? we kind of overwrite it, right?